### PR TITLE
Add function to provide connected host information

### PR DESCRIPTION
**Pull request by jmbents**
Update to add 'libssh2_channel_read_host_info' function to provide
host information which is connected to remote server.

Signed-off-by: Paresh Chaudhary <paresh.chaudhary@rockwellcollins.com>
Signed-off-by: Jared Bents <jared.bents@rockwellcollins.com>

### CODE REVIEW

**Comment by jmbents** [comment]
@bagder Is this something we could get into the next release?

File: src/channel.c [hunk 1]
```diff
@@ -53,6 +53,29 @@
...
+LIBSSH2_API int
+libssh2_channel_read_host_info(LIBSSH2_CHANNEL *channel, unsigned char *shost,
+                               unsigned int *sport, unsigned int *shost_len)
+{
+    int rc = 0;
+
+    if (!channel || !channel->shost)
+        return LIBSSH2_ERROR_BAD_USE;
+
+    memcpy(shost, channel->shost, strlen((char*)channel->shost));
```
Comment (bagder) [inline src/channel.c]: this seems like asking for an overflow. I would prefer the API to pass in a buffer size too to make that impossible. Possibly then return a failure if the buffer was too small.

Comment (jmbents) [reply src/channel.c]: I have updated to add a buffer_size for the user to provide and a check to make sure it is big enough.

**Comment by willco007** [comment]
Could you use the paradigm of calling the function once with NULL to get the required buffer length and call it again with the alloc'd buffer to copy the value in to? That way the client would be much less likely to alloc a buffer that is too small.

File: docs/libssh2_channel_read_host_info.3 [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                                               unsigned int *sport,
+                                               unsigned int *shost_len);
+
+.SH DESCRIPTION
+This function is used to pass shost, sport, and shost_len from library to user space application.
+underlying function \fIlibssh2_channel_read_ex(3)\fP.
+
+\fIchannel\fP - Active channel stream to read from.
+
+\fIshost\fP - Pointer to host to tell the application where the connection originated from.
```
Comment (bagder) [inline docs/libssh2_channel_read_host_info.3]: Please elaborate on this. Where it originated from? Is that a host name? How can a connection originate from a host name? Is it an IP address? Can it be either IPv4 or IPv6? Is the string zero terminated?

File: docs/libssh2_channel_read_host_info.3 [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+\fIshost_buffer_size\fP - Size of buffer allocated for shost.
```
Comment (bagder) [inline docs/libssh2_channel_read_host_info.3]: Is there a recommendation for how large this should be?

File: docs/libssh2_channel_read_host_info.3 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+\fIsport\fP - Pointer to port to tell the application where the connection originated from.
```
Comment (bagder) [inline docs/libssh2_channel_read_host_info.3]: Elaborate. What is a port it originates from. Is this the TCP source port for the connection? If so, why not say so?

File: docs/libssh2_channel_read_host_info.3 [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+\fIshost_len\fP - Pointer to length of shost.
```
Comment (bagder) [inline docs/libssh2_channel_read_host_info.3]: And perhaps explain that the function will set this to the length of that returned string? Is that what it does? If the string is zero terminated, why is this information needed?

File: docs/libssh2_channel_read_host_info.3 [hunk 6]
```diff
@@ -0,0 +1,36 @@
+.TH libssh2_channel_read_host_info 3 "23 Jan 2019" "libssh2 1.8.0" "libssh2 manual"
+.SH NAME
+libssh2_channel_read_host_info - read host information.
+.SH SYNOPSIS
+#include <libssh2.h>
+
+LIBSSH2_API int libssh2_channel_read_host_info(LIBSSH2_CHANNEL *channel,
```
Comment (bagder) [inline docs/libssh2_channel_read_host_info.3]: how about dropping `read` from the name and just make it `libssh2_channel_hostinfo` ? Is it even info about the *host* ?

**Pull request closed without merging**